component/file_cache/file_cache.go

```diff
@@ -626,8 +624,9 @@ func (fc *FileCache) validateStorageError(path string, err error, method string,
 func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	log.Trace("FileCache::DeleteFile : name=%s", options.Name)
 
-	fc.fileLocks.Lock(options.Name)
-	defer fc.fileLocks.Unlock(options.Name)
+	flock := fc.fileLocks.Get(options.Name)
```

Comment:
Should we remove the fileLock from the map here?

Reply:
Similarly on close when all the handle count is 0?

Reply:
Removing the lock item is not safe as it may happen that there are other threads who have taken out the lock item and waiting for the lock. If we delete then those threads will crash during the lock operation. Only way to cleanup I see is to just move them to some queue which periodically cleans it up.